File: SoftUniCourses/JavaScript/VSCode/JavaScript/whileLoop/graduation.js
function graduation(input) {

    let studentName = input[0];
    let grade = 1;
    let sum = 0;
    let repetitions = 0;//i

    let isExcluded = false;
    while (grade <= 12) {

        let currentMark = Number(input[grade]);
        if (grade < 4.00) {
            repetitions++;

            if (repetitions == 2) {

                console.log(`${studentName} has been excluded at ${repetitions} grade`);

                isExcluded = true;
                break;
            }
            continue;
        }
        sum = sum + currentMark;
        grade++;
    }
    if (!isExcluded) {
        let averge = sum / 12;
        console.log(`${studentName} graduated. Average grade: ${sum}`);
    }
}
graduation(["Gosho",
    "5", "5.5", "6", "5.43",
    "5.5", "6", "5.55", "5",
    "6", "6", "5.43", "5"]);

File: SoftUniCourses/SoftUniBasics/whileloops/Graduation.java
package whileloops;

import java.util.Scanner;

public class Graduation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String studentName = scanner.nextLine();

        int grade = 1;
        double sum = 0;// obshata suma ot ot otsenka
        int repetitions =0;
        boolean isExcluded = false;

        while (grade <= 12){
            double currentMark = Double.parseDouble(scanner.nextLine());//ocenka
             if (currentMark < 4){//povtarq klasa
                 repetitions++;
                 if (repetitions == 2){

                     System.out.printf("%s has been excluded at %d grade", studentName,grade);
                     isExcluded = true;
                     break;
                 }
                 continue;
             }
            sum +=currentMark;
            grade++;

        }
        if (!isExcluded){
            double average = sum/ 12;//srednata ocenka
            System.out.printf("%s graduated. Average grade: %.2f", studentName, average);

        }

    }
}
